packages/api-token-auth-client/src/ApiTokenAuthClient.ts

```diff
@@ -1,44 +1,60 @@
-import { AuthState, IAuthClient, SDKError, ERROR_CODES, PACKAGES } from '@8base/utils';
-
-type ApiTokenAuthClientOptions = {
+import {
+  IAuthState,
+  IAuthClient,
+  IStorage,
+  PACKAGES,
+  StorageAPI,
+  throwIfMissingRequiredParameters,
+} from '@8base/utils';
+
+interface IApiTokenAuthClientOptions {
   apiToken: string;
-};
+}
 
 /**
- * Create instacne of the api token auth client
- * @param {string} apiToken Api Token generated in 8base app.
+ * Creates instacne of the api token auth client
  */
 class ApiTokenAuthClient implements IAuthClient {
-  public apiToken: string;
-  public state: {};
+  private storageAPI: StorageAPI<IAuthState>;
+  private apiToken: string;
+
+  constructor(
+    options: IApiTokenAuthClientOptions,
+    storage: IStorage = window.localStorage,
+    storageKey: string = 'auth',
+  ) {
+    throwIfMissingRequiredParameters(['apiToken'], PACKAGES.API_TOKEN_AUTH_CLIENT, options);
 
-  constructor({ apiToken }: ApiTokenAuthClientOptions) {
-    if (!apiToken) {
-      throw new SDKError(ERROR_CODES.MISSING_PARAMETER, PACKAGES.API_TOKEN_AUTH_CLIENT, 'apiToken is required');
-    }
+    const { apiToken } = options;
 
+    this.storageAPI = new StorageAPI<IAuthState>(storage, storageKey);
     this.apiToken = apiToken;
-    this.state = {};
+
+    this.storageAPI.setState({ token: apiToken });
   }
 
-  public getAuthState = async (): Promise<AuthState> => ({
-    token: this.apiToken,
-    ...this.state,
-  });
+  public getState(): IAuthState {
+    return {
+      ...this.storageAPI.getState(),
+      token: this.apiToken,
+    };
+  }
 
-  public setAuthState = async (state: AuthState): Promise<void> => {
-    this.state = {
-      ...this.state,
+  public setState(state: IAuthState): void {
+    this.storageAPI.setState({
       ...state,
       token: this.apiToken,
-    };
-  };
+    });
+  }
 
-  public purgeAuthState = async (): Promise<void> => {
-    this.state = {};
-  };
+  public purgeState(): void {
+    this.storageAPI.purgeState();
+    this.storageAPI.setState({ token: this.apiToken });
```

Comment:
@vorobeez why do u set token in the purge function?

Reply:
'Cause the idea behind api-token client is that you can't rewrite token property.